magento/magento2
# Replaced the page refresh from the gift option with a XHR request

### Description
Replaced the page refresh with a XHR request. Also added blockloaders for usability purposes. Updating and deleting gift messages will go way smoother now. Also refactored some code with uses. 

### Fixed Issues (if relevant)

1. magento/magento2#9206: Apply Shopping Cart Gift Options without page reload

### Manual testing scenarios
1. Install Magento latest develop version
2. Activate Gift Option from the backend (Stores>Configuration>Sales>Sales>Gift Options)
3. add product to cart
4. add gift message to product
5. delete/update gift message
6. add gift option to cart
7. delete/update gift message

Expected result:
After editing or deleting the message or option, there should not be a page refresh but a block loader on the section, and then the edited content. 

### Contribution checklist
